examples/simple_config_with_hooks/main.rs

```diff
@@ -170,7 +171,13 @@ fn main() {
     // reference implementation of this trait that uses the XCB library to communicate with the X
     // server. You are free to provide your own implementation if you wish, see xconnection.rs for
     // details of the required methods and expected behaviour.
-    let conn = XcbConnection::new().unwrap();
+    let mut conn = XcbConnection::new().unwrap();
+
+    let modifier_map = modifiers_from_xmodmap();
+
+    if let Some(&numlock_mask) = modifier_map.get("Num_Lock") {
+        conn.set_numlock_mask(numlock_mask);
```

Comment:
I really would prefer this to be a parameter in `new` rather than a method that mutates `XcbConnection` after the fact as mentioned previously.